doc/classes/NetworkedMultiplayerCustom.xml

```diff
@@ -6,6 +6,70 @@
+		    get_tree().network_peer = custom_peer
+
+		    # Start connecting via real transport (ex. Steam peer-to-peer, PlayFab, etc).
+		    real_multiplayer_client = _connect_to_real_multiplayer()
+		    real_multiplayer_client.connect("connection_succeeded", self, "_on_real_multiplayer_connection_succeeded")
+		    real_multiplayer_client.connect("connection_failed", self, "_on_real_multiplayer_connection_failed")
+		    real_multiplayer_client.connect("peer_connected", self, "_on_real_multiplayer_peer_connected")
+		    real_multiplayer_client.connect("peer_disconnected", self, "_on_real_multiplayer_peer_disconnected")
+		    real_multiplayer_client.connect("recieved_message", self, "_on_real_multiplayer_received_message")
+
+		    # Intercept any RPC's sent locally.
+		    custom_peer.connect("packet_generated", self, "_on_custom_peer_packet_generated")
+
+		func _on_real_multiplayer_connection_succeeded():
+		    # Once connection is formed, assign a peer id somehow. If the real transport
+		    # doesn't have a compatible id, you'll need to generate your own ids, and map
+		    # them to the ids used by the real transport (the same mapping should be used
+		    # by all peers).
+		    #
+		    # Also, if you want Godot to treat this peer as the "host" or "server" (per
+		    # `get_tree().is_network_server()`) then use the peer id of 1.
+		    var my_id = real_multiplayer_client.get_my_id()
+		    custom_peer.initialize(my_id)
+		    custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_CONNECTED)
+
+		func _on_real_multiplayer_connection_failed():
+		    custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_DISCONNECTED)
+
+		func _on_real_multiplayer_peer_connected(peer_id):
+		    # Let Godot know a peer connected. Again, you may need to map between Godot ids
+		    # and the ids used by the real multiplayer client.
+		    custom_peer.emit_signal("peer_connected", peer_id)
+
+		func _on_real_multiplayer_peer_disconnected(peer_id):
+		    # Same notes as above.
+		    custom_peer.emit_signal("peer_disconnected", peer_id)
+
+		func _on_custom_peer_packet_generated(peer_id, buffer, transfer_mode):
+		    # Send any packets generated locally (ex RPCs) to the requested peer
+		    # using the real multiplayer client.
+		    if transfer_mode == NetworkedMultiplayerPeer.TRANSFER_MODE_RELIABLE:
+		        real_multiplayer_client.send_reliable(peer_id, {
+		            'from_peer_id': custom_peer.get_unique_id(),
+		            'data': buffer,
+		        })
+		    else:
+		        real_multiplayer_client.send_unreliable(peer_id, {
+		            'from_peer_id': custom_peer.get_unique_id(),
+		            'data': buffer,
+		        })
+
+		func _on_real_multiplayer_received_message(message):
+		    # Take any packets we recieved from other peers, and deliver them locally.
+		    custom_peer.deliver_packet(message['data'], message['from_peer_id'])
+		[/codeblock]
```

Comment:
Here's that C# translation. I'll give it an actual go in my project here in a minute and make any tweaks if I see any that would be helpful.
```suggestion
	[codeblocks]
        [gdscript]
		extends Node

		var custom_peer := NetworkedMultiplayerCustom.new()
		var real_multiplayer_client

		func join_multiplayer():
		    custom_peer.set_connection_status(NetworkedMultiplayerPeer.CONNECTION_CONNECTING)
		    get_tree().network_peer = custom_peer

		    # Start connecting via real transport (ex. Steam peer-to-peer, PlayFab, etc).
		    real_multiplayer_client = _connect_to_real_multiplayer()
		    real_multiplayer_client.connect("connection_succeeded", self, "_on_real_multiplayer_connection_succeeded")
		    real_multiplayer_client.connect("connection_failed", self, "_on_real_multiplayer_connection_failed")
		    real_multiplayer_client.connect("peer_connected", self, "_on_real_multiplayer_peer_connected")
		    real_multiplayer_client.connect("peer_disconnected", self, "_on_real_multiplayer_peer_disconnected")
		    real_multiplayer_client.connect("recieved_message", self, "_on_real_multiplayer_received_message")

		    # Intercept any RPC's sent locally.
		    custom_peer.connect("packet_generated", self, "_on_custom_peer_packet_generated")

		func _on_real_multiplayer_connection_succeeded():
		    # Once connection is formed, assign a peer id somehow. If the real transport
		    # doesn't have a compatible id, you'll need to generate your own ids, and map
		    # them to the ids used by the real transport (the same mapping should be used
		    # by all peers).
		    #
		    # Also, if you want Godot to treat this peer as the "host" or "server" (per
		    # `get_tree().is_network_server()`) then use the peer id of 1.
		    var my_id = real_multiplayer_client.get_my_id()
		    custom_peer.initialize(my_id)
		    custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_CONNECTED)

		func _on_real_multiplayer_connection_failed():
		    custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_DISCONNECTED)

		func _on_real_multiplayer_peer_connected(peer_id):
		    # Let Godot know a peer connected. Again, you may need to map between Godot ids
		    # and the ids used by the real multiplayer client.
		    custom_peer.emit_signal("peer_connected", peer_id)

		func _on_real_multiplayer_peer_disconnected(peer_id):
		    # Same notes as above.
		    custom_peer.emit_signal("peer_disconnected", peer_id)

		func _on_custom_peer_packet_generated(peer_id, buffer, transfer_mode):
		    # Send any packets generated locally (ex RPCs) to the requested peer
		    # using the real multiplayer client.
		    if transfer_mode == NetworkedMultiplayerPeer.TRANSFER_MODE_RELIABLE:
		        real_multiplayer_client.send_reliable(peer_id, {
		            'from_peer_id': custom_peer.get_unique_id(),
		            'data': buffer,
		        })
		    else:
		        real_multiplayer_client.send_unreliable(peer_id, {
		            'from_peer_id': custom_peer.get_unique_id(),
		            'data': buffer,
		        })

		func _on_real_multiplayer_received_message(message):
		    # Take any packets we recieved from other peers, and deliver them locally.
		    custom_peer.deliver_packet(message['data'], message['from_peer_id'])
        [/gdscript]
        [csharp]
        using Godot;

        public class CustomLobby : Node
        {

            NetworkedMultiplayerCustom customPeer;
            RealMultiplayerClient realMultiplayerClient;

            void JoinMultiplayer()
            {
                customPeer.SetConnectionStatus(NetworkedMultiplayerPeer.CONNECTION_CONNECTING);
                GetTree().NetworkPeer = customPeer;

                // Start connecting via real transport (ex. Steam peer-to-peer, PlayFab, etc).
                realMultiplayerClient = _ConnectToRealMultiplayer();

                realMultiplayerClient.Connect("connection_succeeded", this, nameof(_OnRealMultiplayerConnectionSucceeded));

                realMultiplayerClient.Connect("connection_failed", this, nameof(_OnRealMultiplayerConnectionFailed));

                realMultiplayerClient.Connect("peer_connected", this, nameof(_OnRealMultiplayerPeerConnected));

                realMultiplayerClient.Connect("peer_disconnected", this, nameof(_OnRealMultiplayerPeerDisconnected));

                realMultiplayerClient.Connect("recieved_message", this, nameof(_OnRealMultiplayerReceivedMessage));

                // Intercept any RPC's sent locally.
                customPeer.Connect("packet_generated", this, nameof(_OnCustomPeerPacketGenerated));
            }

            void _OnRealMultiplayerConnectionSucceeded()
            {
                // Once connection is formed, assign a peer id somehow. If the real transport
                // doesn't have a compatible id, you'll need to generate your own ids, and map
                // them to the ids used by the real transport (the same mapping should be used
                // by all peers).
                //
                // Also, if you want Godot to treat this peer as the "host" or "server" (per
                // `GetTree().IsNetworkServer()`) then use the peer id of 1.
                var myId = realMultiplayerClient.GetMyId();

                customPeer.Initialize(myId);

                customPeer.SetConnectionStatus(NetworkedMultiplayerCustom.CONNECTION_CONNECTED);
            }

            void _OnRealMultiplayerConnectionFailed()
            {
                customPeer.SetConnectionStatus(NetworkedMultiplayerCustom.CONNECTION_DISCONNECTED);
            }

            void _OnRealMultiplayerPeerConnected(int peerId)
            {
                // Let Godot know a peer connected. Again, you may need to map between Godot ids
                // and the ids used by the real multiplayer client.
                customPeer.EmitSignal("peer_connected", peerId);
            }

            void _OnRealMultiplayerPeerDisconnected(int peerId)
            {
                // Same notes as above.
                customPeer.EmitSignal("peer_disconnected", peerId);
            }

            void _OnCustomPeerPacketGenerated(int peerId, byte[] buffer, int transferMode)
            {
                // Send any packets generated locally (ex RPCs) to the requested peer
                // using the real multiplayer client.
                if (transferMode == NetworkedMultiplayerPeer.TRANSFER_MODE_RELIABLE)
                {

                    realMultiplayerClient.SendReliable(peerId, new RealPacket()
                    {
                        fromPeerId = customPeer.GetNetworkUniqueId(),
                        data = buffer
                    });
                }

                else
                {

                    realMultiplayerClient.SendUnreliable(peerId, new RealPacket()
                    {
                        fromPeerId = customPeer.GetNetworkUniqueId(),
                        data = buffer
                    });
                }
            }

            void _OnRealMultiplayerReceivedMessage(RealPacket message)
            {
                // Take any packets we recieved from other peers, and deliver them locally.
                customPeer.DeliverPacket(message.data, message.fromPeerId);
            }
        }
        [/csharp]
		[/codeblock]
```

Reply:
Ew... well that suggestion preview is illegible. I think I'll just make a PR off of this fork for you to review.

Reply:
Awesome, thanks! Let's wait for the GDScript version to get reviewed by the docs team, and once it's merged, you could do the C# version as a separate PR to Godot (with any changes from the GDScript review)? And that way you'd get commit in Godot in your name as well ;-)

Reply:
👍